staticaddr/withdraw/manager.go

```diff
@@ -325,37 +426,29 @@ func (m *Manager) WithdrawDeposits(ctx context.Context,
 		d.Unlock()
 	}
 
-	// Transition the deposits to the withdrawing state. This updates each
-	// deposits withdrawal address. If a transition fails, we'll return an
-	// error and abort the withdrawal. An error in transition is likely due
-	// to an error in the state machine. The already transitioned deposits
-	// should be reset to the Deposit state after a restart.
+	// Add the new withdrawal tx to the finalized withdrawals to republish
+	// it on block arrivals.
+	m.mu.Lock()
+	m.finalizedWithdrawalTxns[finalizedTx.TxHash()] = finalizedTx
+	m.mu.Unlock()
+
+	// Transition the deposits to the withdrawing state. If the user fee
+	// bumped a withdrawal this results in a NOOP transition.
 	err = m.cfg.DepositManager.TransitionDeposits(
 		ctx, deposits, deposit.OnWithdrawInitiated, deposit.Withdrawing,
 	)
 	if err != nil {
 		return "", "", err
 	}
 
-	err = m.publishFinalizedWithdrawalTx(ctx, finalizedTx)
-	if err != nil {
-		return "", "", err
-	}
-
-	withdrawalPkScript, err := txscript.PayToAddrScript(withdrawalAddress)
-	if err != nil {
-		return "", "", err
-	}
-
-	err = m.handleWithdrawal(
-		ctx, deposits, finalizedTx.TxHash(), withdrawalPkScript,
-	)
-	if err != nil {
-		return "", "", err
+	// Update the deposits in the database.
+	for _, d := range deposits {
+		err = m.cfg.DepositManager.UpdateDeposit(ctx, d)
+		if err != nil {
+			return "", "", err
```

Comment:
I propose to use `fmt.Errorf` to attach an error message to simplify debugging.

Reply:
done